packages/aws-cdk-lib/cx-api/lib/features.ts

```diff
@@ -1034,6 +1036,41 @@ export const FLAGS: Record<string, FlagInfo> = {
     introducedIn: { v2: 'V2NEXT' },
     recommendedValue: true,
   },
+
+  //////////////////////////////////////////////////////////////////////
+  [ENABLE_ALBV2_EXTERNALAPPLICATIONLISTENER_ADDTARGETGROUP_TO_ADDACTION_MIGRATION]: {
+    type: FlagType.VisibleContext,
+    summary: 'When enabled, you can migrate from the \`addTargetGroups()\` method of declaring a \`ListenerRule\` to the \`addAction()\` method, without changing the logicalId and replacing your resource.',
+    detailsMd: `
+      When migrating from a less complex to a more complex use of ALB,
+      you will eventually need features not available in the \`addTargetGroups()\` convenience method.
+      In this case you will want to use the \`addAction()\` method.
+      Without this feature being activated, the change will also add a \`Rule\` suffix to the logicalId of your \`ListnerRule\`,
+      causing CloudFormation to attempt to replace the resource.
+      Since ListenerRules have a unique priority,
+      CloudFormation will always fail when generating the new ListenerRule.
+
+      Setting this feature flag will cause the \`addAction()\` method to not add the \`Rule\` suffix on the logicalId,
+      enabling a migration from the \`addTargetGroups()\` method.
+
+      For new resources, see instead the ${ENABLE_ALBV2_EXTERNALAPPLICATIONLISTENER_ADDTARGETGROUP_CONSISTENT_LOGICALID} flag.
+    `,
+    introducedIn: { v2: 'V2NEXT' },
+    recommendedValue: false,
+  },
+
+  //////////////////////////////////////////////////////////////////////
+  [ENABLE_ALBV2_EXTERNALAPPLICATIONLISTENER_ADDTARGETGROUP_CONSISTENT_LOGICALID]: {
```

Comment:
I think that just having the other feature flag is sufficient. I can only think of a couple of scenarios where we would want to have this second feature flag:
1. Users want to keep `Rule` in the logicalId
2. Eventually get everyone to use this feature flag and then we can remove the feature flags altogether and migrate to add `Rule` to the `addTargetGroups()` case which I can't see happening. 

Could you explain why you feel this second feature flag is necessary? My main worry is that it will cause confusion with the first feature flag.

Reply:
Users don't care about the contents of logicalIds until it causes them problems. This flag allows anyone who actually does care to be future friendly, while allowing the vast majority of users who don't care to continue not caring. When the logicalId changes, Cfn will attempt to replace the resource and fail. There is no way I can see to make this change painless for existing users. That said, maybe we should invert the logic here: make things consistent by default (breaking existing users), but provide a feature flag which allows legacy behaviour so existing users can punt until vNext. That said, this isn't what's blocking us, and maybe we should split this into two separate PRs?

Reply:
I did that. The consistency stuff can be a separate PR. We just need compatibility to get us unblocked.